File: samples/vulnerable_files/sqli.py
# Intentionally vulnerable Python file for testing SQL Injection
import sqlite3


def get_user_profile(request):
    # Retrieve user input from the request object
    user_id = request.GET.get("id")

    # Vulnerable: direct f-string interpolation of user input into a SQL query
    query = "SELECT * FROM users WHERE id = ?"

    conn = sqlite3.connect("users.db")
    cursor = conn.cursor()

    # Vulnerable execute sink
    cursor.execute(query, (user_id,))
    return cursor.fetchone()

File: samples/vulnerable_files/SQLi.java
// Intentionally vulnerable Java file for testing SQL Injection
import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.Statement;

public class SQLi {
    public void executeQuery(Connection conn, String userInput) throws Exception {
        // Statement stmt = conn.createStatement();

        // Vulnerable: raw string concatenation of untrusted input in SQL statement
        String query = "SELECT * FROM users WHERE username = ?";

        // Vulnerable executeQuery sink
        PreparedStatement pstmt = conn.prepareStatement(query);
        pstmt.setString(1, userInput);
        ResultSet rs = pstmt.executeQuery();
        while (rs.next()) {
            System.out.println(rs.getString("username"));
        }
    }
}
